contrib/externs/angular_ui_router.js

```diff
@@ -55,6 +54,93 @@ ui.router.$state;
 
 
 /**
+ * @type {ui.router.State}
+ */
+ui.router.$state.current;
+
+
+/**
+ * @type {Object}
+ */
+ui.router.$state.params;
+
+
+/**
+ * @type {?angular.$q.Promise}
+ */
+ui.router.$state.transition;
+
+
+/**
+ * @param {?string|Object=} stateOrName
+ * @param {?string|Object=} context
+ * @return {Object|Array}
+ */
+ui.router.$state.get = function(stateOrName, context) {};
+
+
+/**
+ * @param {string} to
+ * @param {Object=} params
+ * @param {(ui.router.$state.GoOptions_|Object)=} options
+ * @returns {angular.$q.Promise}
+ */
+ui.router.$state.go = function(to, params, options) {};
+
+
+/**
+ * @param {?string|Object} stateOrName
+ * @param {Object=} params
+ * @param {Object=} options
+ * @returns {string} compiled state url
+ */
+ui.router.$state.href = function(stateOrName, params, options) {};
+
+
+/**
+ * @param {?string} stateOrName
+ * @param {Object=} params
+ * @param {Object=} options
+ */
+ui.router.$state.includes = function(stateOrName, params, options) {};
+
+
+/**
+ * @param {?string|Object} stateOrName
+ * @param {Object=} params
+ * @param {Object=} options
+ * @returns {boolean}
+ */
+ui.router.$state.is = function(stateOrName, params, options) {};
+
+
+/**
+ * @returns {angular.$q.Promise}
+ */
+ui.router.$state.reload = function() {};
+
+
+/**
+ * @param {string} to
+ * @param {Object=} toParams
+ * @param {Object=} options
+ */
+ui.router.$state.transitionTo = function(to, toParams, options) {};
+
+
+/**
+ * @typedef {{
+ *   location: (boolean|string|undefined),
+ *   inherit: (boolean|undefined),
+ *   relative: (Object|undefined),
+ *   notify: (boolean|undefined),
+ *   reload: (boolean|undefined)
+ * }}
+ */
+ui.router.$state.GoOptions_;
```

Comment:
I think JSCompiler used to have a bug that it would (silently!) not understand type definitions in externs that were used before they were defined. If you're not 100% sure this works as is, please move up front before any use sites.